PVTPackage/source/MultiphaseSystem/MultiphaseSystemProperties/BlackOilDeadOilMultiphaseSystemProperties.cpp

```diff
@@ -23,19 +23,19 @@
 namespace PVTPackage
 {
 
-BlackOilDeadOilMultiphaseSystemProperties::BlackOilDeadOilMultiphaseSystemProperties( std::size_t nComponents )
+BlackOilDeadOilMultiphaseSystemProperties::BlackOilDeadOilMultiphaseSystemProperties( std::vector< pvt::PHASE_TYPE > const & phases )
   :
-  FactorMultiphaseSystemProperties( { pvt::PHASE_TYPE::OIL, pvt::PHASE_TYPE::GAS, pvt::PHASE_TYPE::LIQUID_WATER_RICH }, nComponents )
+  FactorMultiphaseSystemProperties( phases, phases.size() )
```

Comment:
I'm really concerned by this line.
I don't get all the physics, but I though that `phases` and `nComponents` were two distinct things.

When you write
```c++
FactorMultiphaseSystemProperties( phases, phases.size() )
```
You provide redundant information, which is error prone since it allows any dev to provide non consistent input.
Are those two parameters meant to be coordinated together?
Specifically for `BlackOil` and `DeadOil`?
What about the other cases (`NegativeTwoPhase`, `FreeWater`, `Trivial`)?
Would it make sense to provide `phase` and and a number of `components` not equal to the number of `phases`? If it does not make any sense, we should change the interface into `FactorMultiphaseSystemProperties( phases )` only...

Reply:
Yes, I realize now that I wanted to add a comment about this in the code I just committed but I forgot. 
- For `DeadOil` and `BlackOil`, the number of phases and the number of components will always be equal (if they are not equal we should error out). 
- For the other cases, they can be different. 

To me it makes sense to keep:
```
FactorMultiphaseSystemProperties( phases, nComponents )
```
to handle all cases.

Reply:
OK, fine. So just add a comment at the appropriate location (find it yourself 😝 🤣 )

Reply:
Done